pkg/pushsync/pushsync.go

```diff
@@ -289,17 +238,22 @@ func (ps *PushSync) PushChunkToClosest(ctx context.Context, ch swarm.Chunk) (r *
 		if err != nil {
 			ps.metrics.TotalErrors.Inc()
 			lastErr = fmt.Errorf("new stream for peer %s: %w", peer.String(), err)
-			logger.Debugf("pushsync-push: %v", lastErr)
+			logger.Errorf("pushsync-push: %v", lastErr)
 			continue
 		}
 		deferFuncs = append(deferFuncs, func() { go streamer.FullClose() })
```

Comment:
why is this in defer?
we can just Release, FullClose and Reset right here in case of error, there is nothing to defer, 
what am i not getting?

Reply:
not sure what you mean? should we just stop using `defer` pattern in general?